airbyte-integrations/connectors/source-mysql/src/main/resources/spec.json

```diff
@@ -54,11 +54,140 @@
         "default": true,
         "order": 6
       },
+      "ssl_mode": {
+        "title": "SSL modes",
+        "description": "SSL connection modes. \n <ul><li><b>disable</b> - Use to make a general connection without SSL.</li>\n <li><b>preferred</b> - Automatically attempt SSL connection. If the MySQL server does not support SSL, continue with a regular connection.</li>\n <li><b>required</b> - Always connect with SSL. If the MySQL server doesn’t support SSL, the connection will not be established. Certificate Authority (CA) and Hostname are not verified.</li>\n  <li><b>verify-ca</b> - Always connect with SSL. Verifies CA, but allows connection even if Hostname does not match.</li>\n  <li><b>Verify Identity</b> -Always connect with SSL. Verify both CA and Hostname.</li></ul>\n Read more <a href=\"https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-using-ssl.html\"> in the docs</a>.",
+        "type": "object",
+        "order": 7,
+        "oneOf": [
+          {
+            "title": "disable",
+            "description": "Disable SSL.",
+            "required": ["mode"],
+            "properties": {
+              "mode": {
+                "type": "string",
+                "const": "disable",
+                "enum": ["disable"],
+                "default": "disable",
+                "order": 0
+              }
+            }
+          },
+          {
+            "title": "preferred",
+            "description": "Preferred SSL mode.",
+            "required": ["mode"],
+            "properties": {
+              "mode": {
+                "type": "string",
+                "const": "preferred",
+                "enum": ["preferred"],
+                "default": "preferred",
+                "order": 0
+              }
+            }
+          },
+          {
+            "title": "required",
+            "description": "Require SSL mode.",
+            "required": ["mode"],
+            "properties": {
+              "mode": {
+                "type": "string",
+                "const": "required",
+                "enum": ["required"],
+                "default": "required",
+                "order": 0
+              }
+            }
+          },
+          {
+            "title": "Verify CA",
+            "description": "Verify CA SSL mode.",
+            "required": ["mode"],
+            "properties": {
+              "mode": {
+                "type": "string",
+                "const": "verify_ca",
+                "enum": ["verify_ca"],
+                "default": "verify_ca",
+                "order": 0
+              },
+              "ca_certificate": {
+                "type": "string",
+                "title": "CA certificate",
+                "description": "CA certificate",
+                "airbyte_secret": true,
+                "multiline": true,
+                "order": 1
+              },
+              "client_key_password": {
```

Comment:
this field is optional and the user can leave it empty, in which case the password for the keystore will be generated automatically. Yes, this is not mandatory for trustkeystore where the root certificate will be stored, but I suggest leaving everything as it is, especially since this field is not mandatory to be filled by the user.

Reply:
Fair enough. I just have a concern about reusing this password when both keystrokes are necessary. That is when client certificate is also provided.